### PR TITLE
add setting for num of recent items on welcome page

### PR DESCRIPTION
**Pull request by sbatten**
This PR adds a new setting for users to configure the number of recent items displayed on the welcome page.

### CODE REVIEW

File: src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts [hunk 1]
```diff
@@ -311,7 +311,7 @@ class WelcomePage extends Disposable {
 				return;
 			}
 			const moreRecent = ul.querySelector('.moreRecent')!;
-			const workspacesToShow = workspaces.slice(0, 5);
+			const workspacesToShow = workspaces.slice(0, this.configurationService.getValue<number>('welcome.numOpenRecentEntries') || 5);
```
Comment (sana-ajani) [inline src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts]: you have to add the setting!

Comment (sbatten) [reply src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts]: oh yea!

Comment (sana-ajani) [inline src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts]: REGISTER THE SETTING!

Comment (sana-ajani) [inline src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts]: add the setting

Comment (sbatten) [reply src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts]: ok thanks

Comment (sbatten) [reply src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts]: helllo

Comment (sbatten) [reply src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts]: thanks

**Comment by Tyriar** [comment]
Closing, this was used as a demo at OSCON

**Pull request closed without merging**